### ISSUE TITLE
Issue with Eval Set

### ISSUE THREAD

**Issue by Dhru1001** [Hope1337/YOWOv3#69]
So the issue is tried training my model on Ava changed it label's since my own dataset is in the same format now i would like to clarify a few things on these :

1. My csv is ava styled : Where the only notable difference is instead of Timestamp in csv its the frame number  
2. During training what is the suitable amount of epochs recommended 
3. Now to the main issue - my lables start from 0 to 11 but pbtxt file as mentioned has to be 1 based so there is always a mismatch in '' the following 11 label has no ground truth "
4.  Eval map score is always around 0.005 / 0.002 but the loss function goes from 6 to 2 in 9epochs, 25epochs the same 
5. Even tried for the okutama dataset the same results map score low

**Comment by Hope1337** [comment]
Hi, I have just had free time to continue this project.
1. Frame number is used to extract the correct image (i.e. in correct order), so make sure that it's aligned.
2. It depends on your size of dataset, if your data size is comparable to AVA, then 9 epoch should be nice.
3. I don't follow, please explain the problem a little bit more.
4. I don't have enough context to diagnose the problem, so more information is preferable.
5. Same with 4.

**Comment by Dhru1001** [comment]
@Hope1337  Hello , Thank you  for your quick response.

1.  Yes The frame number is correctly matched to the image in write order 
2. I will share the config with you 
3. So in the config there is a detection argument where we enter the pbtxt file which are nothing but the label's from 1-12 in my case so now but during the training i had my label's starting from  0-11 will share config for better understating 
4. While training we have see that the loss on termial is starting from 6.0 and gradually goes down to 2.0 within the first 9 epochs 


BACKBONE2D:
  YOLOv8:
    ver : d

    PRETRAIN:
      n : weights/backbone2D/YOLOv8/v8_n.pth
      s : weights/backbone2D/YOLOv8/v8_s.pth
      m : weights/backbone2D/YOLOv8/v8_m.pth
      l : weights/backbone2D/YOLOv8/v8_l.pth
      x : weights/backbone2D/YOLOv8/v8_x.pth
      d : weights/backbone2D/YOLOv8/visdrone.pth

BACKBONE3D:

  MOBILENET:
    width_mult: 2.0

    PRETRAIN:
      width_mult_0.5x  : weights/backbone3D/mobilenet/kinetics_mobilenet_0.5x_RGB_16_best.pth
      width_mult_1.0x  : weights/backbone3D/mobilenet/kinetics_mobilenet_1.0x_RGB_16_best.pth
      width_mult_1.5x  : weights/backbone3D/mobilenet/kinetics_mobilenet_1.5x_RGB_16_best.pth
      width_mult_2.0x  : weights/backbone3D/mobilenet/kinetics_mobilenet_2.0x_RGB_16_best.pth

  MOBILENETv2:
    width_mult: 1.0

    PRETRAIN:
      width_mult_0.2x  : weights/backbone3D/mobilenetv2/kinetics_mobilenetv2_0.2x_RGB_16_best.pth
      width_mult_0.45x : weights/backbone3D/mobilenetv2/kinetics_mobilenetv2_0.45x_RGB_16_best.pth
      width_mult_0.7x  : weights/backbone3D/mobilenetv2/kinetics_mobilenetv2_0.7x_RGB_16_best.pth
      width_mult_1.0x  : weights/backbone3D/mobilenetv2/kinetics_mobilenetv2_1.0x_RGB_16_best.pth

  SHUFFLENET:
    width_mult: 2.0

    PRETRAIN:
      width_mult_0.5x  : weights/backbone3D/shufflenet/kinetics_shufflenet_0.5x_G3_RGB_16_best.pth
      width_mult_1.0x  : weights/backbone3D/shufflenet/kinetics_shufflenet_1.0x_G3_RGB_16_best.pth
      width_mult_1.5x  : weights/backbone3D/shufflenet/kinetics_shufflenet_1.5x_G3_RGB_16_best.pth
      width_mult_2.0x  : weights/backbone3D/shufflenet/kinetics_shufflenet_2.0x_G3_RGB_16_best.pth

  SHUFFLENETv2:
    width_mult: 2.0

    PRETRAIN:
      width_mult_0.25x : weights/backbone3D/shufflenetv2/kinetics_shufflenetv2_0.25x_RGB_16_best.pth
      width_mult_1.0x  : weights/backbone3D/shufflenetv2/kinetics_shufflenetv2_1.0x_RGB_16_best.pth
      width_mult_1.5x  : weights/backbone3D/shufflenetv2/kinetics_shufflenetv2_1.5x_RGB_16_best.pth
      width_mult_2.0x  : weights/backbone3D/shufflenetv2/kinetics_shufflenetv2_2.0x_RGB_16_best.pth

  I3D:
    PRETRAIN:
      default: weights/backbone3D/I3D/rgb_imagenet.pth

  RESNET:
    ver : 101

    PRETRAIN:
      ver_18 : weights/backbone3D/resnet/kinetics_resnet_18_RGB_16_best.pth
      ver_50 : weights/backbone3D/resnet/kinetics_resnet_50_RGB_16_best.pth
      ver_101: weights/backbone3D/resnet/kinetics_resnet_101_RGB_16_best.pth

  RESNEXT:
    ver : 101

    PRETRAIN:
      ver_101 : weights/backbone3D/resnext/resnext-101-kinetics.pth

LOSS:
  TAL:
    top_k  : 10
    alpha  : 0.5
    beta   : 6.0
    radius : 2.5
    soft_label : False

    scale_cls_loss : 0.5
    scale_box_loss : 7.5
    scale_dfl_loss : 1.5

  SIMOTA:
    top_k  : 10
    gamma  : 0.5
    radius : 2.5
    mode   : unbalance
    soft_label : True

    dynamic_k     : False
    dynamic_top_k : 10

    scale_cls_loss : 0.5
    scale_box_loss : 5.0
    scale_dfl_loss : 0.5

config_path       : /home1/om/YOWOv3_new/config/cf2/okutama.yaml
dataset           : okutama
loss              : simota
active_checker    : True
num_classes       : 12
backbone2D        : yolov8
backbone3D        : resnet
fusion_module     : CFAM
mode              : decoupled
interchannels     : [256, 256, 256]
pretrain_path     : /home1/om/YOWOv3_new/model_checkpoint/sept16/ema_epoch_9.pth
data_root         : /home1/om/YOWOv3_new/test_dataset
img_size          : 640
clip_length       : 16
batch_size        : 16
num_workers       : 6
acc_grad          : 16
lr                : 0.0001
weight_decay      : 0.0005
max_step_warmup   : 500
adjustlr_schedule : [3, 4, 5, 6]
max_epoch         : 9
lr_decay          : 0.5
save_folder       : model_checkpoint/sept16
sampling_rate     : 1
labelmap          : /home1/om/OKutama_dataset/okutama_action_list.pbtxt
groundtruth       : /home1/om/OKutama_dataset/okutama_test_ava_style.csv
detections        : ava_result/aug29/vis_drone_okutama_null/ava_predicted_file.csv
iou_thresh: 0.5


class_ratio:
  0: 0.26428883172840306
  1: 0.02851730446006556
  2: 0.09331686616088923
  3: 0.2792708633193734
  4: 0.020141332912368023
  5: 0.1287615262468776
  6: 0.029208182822758128
  7: 0.014088718439209204
  8: 0.0586392296334816
  9: 0.010285173044600656
  10: 0.060219706748135836
  11: 0.013262264483837717
idx2name:
  0: Walking
  1: Running
  2: Sitting
  3: Standing
  4: Lying
  5: Carrying
  6: Calling
  7: Hugging
  8: Reading
  9: Drinking
  10: Pushing/Pulling
  11: Hand Shaking
train_class_count:
  0: 142305
  1: 15355
  2: 50246
  3: 150372
  4: 10845
  5: 69331
  6: 15727
  7: 7586
  8: 31574
  9: 5538
  10: 32425
  11: 7141
